star-dash/star-dash/GameEngine/Systems/Complex/CollisionSystem.swift

```diff
@@ -104,6 +104,12 @@ class CollisionSystem: System {
     }
 
     private func handlePlayerObstacleContactEvent(event: PlayerObstacleContactEvent) {
+        if let hookOwnerComponent = entityManager
+                                    .components(ofType: GrappleHookOwnerComponent.self)
+                                    .first(where: { $0.ownerPlayerId == event.playerId }) {
+            dispatcher?.add(event: ReleaseGrappleHookEvent(using: hookOwnerComponent.entityId))
```

Comment:
I'm quite conflicted about having any grapple-related logic here, and generally the purpose of collision system. It feels like `CollisionSystem` may fall into a "god" class problem where it's handling too many logic. For example , once we start having more features that rely on `PlayerObstacleContactEvent`, this single event listener will be doing way too much.

TLDR, I think this should be in `GrappleHookSystem` since its a grapple related event and we may want to start considering to phase out `CollisionSystem`

Wondering how @ruihan00 and @jasonqiu212 feels about this?

Reply:
Sound good to me. Move logic to GrappleHookSystem